<a href="https://colab.research.google.com/github/ElofssonLab/kb8029-book/blob/main/notebooks/day05-discussion-3.ipynb" style="display:inline-block;padding:10px 18px;background-color:#F9AB00;color:#000000;font-weight:bold;text-decoration:none;border-radius:6px;font-family:sans-serif;font-size:14px;">&#9654;&nbsp; Open in Google Colab</a>

# Day 5 — In-class discussion problem (3 of 3)

Work this out **by hand in your group first** — then run the code cell to check your answer before presenting.

## One protein, three domains, three very different scores

The book page's Pfam example used *HBB* and found one domain (Globin, score ~1.3&times;10&#8315;&sup2;&#8319;). **BRCA1** (UniProt `P38398`, the breast-cancer susceptibility gene) is a much larger protein with **three** separate Pfam domain hits at very different confidence levels.

1. Without running anything, what would it mean biologically for one domain on the same protein to have a far more extreme (smaller) score than another?
2. Predict: does a weaker (less extreme) domain score mean that domain is less real, or just less unambiguously detectable from sequence alone?
3. Run the cell below, then rank the three domains from most to least confidently detected.

In [1]:
import requests

r = requests.get("https://www.ebi.ac.uk/interpro/api/entry/pfam/protein/uniprot/P38398/", timeout=10)
r.raise_for_status()
results = r.json()["results"]

print(f"{'Pfam ID':<10}{'Name':<45}{'Region':<15}{'Score':>12}")
for entry in results:
    meta = entry["metadata"]
    loc = entry["proteins"][0]["entry_protein_locations"][0]
    frag = loc["fragments"][0]
    region = f"{frag['start']}-{frag['end']}"
    print(f"{meta['accession']:<10}{meta['name']:<45}{region:<15}{loc['score']:>12}")

Pfam ID   Name                                         Region                Score
PF00097   Zinc finger, C3HC4 type (RING finger)        24-64               1.6e-06
PF00533   BRCA1 C Terminus (BRCT) domain               1645-1723           4.6e-07
PF12820   Serine-rich domain associated with BRCT      345-508             2.8e-77


**Discussion point:** all three domains are equally "real" — BRCA1 genuinely has a RING finger domain, a BRCT domain, and the serine-rich region associated with it. What differs is how *unambiguous* the sequence signal is: the serine-rich/BRCT-associated domain (PF12820) scores extremely small (~2.8&times;10&#8315;&sup7;&sup7;), an extraordinarily confident call, while the RING finger domain (PF00097) scores a comparatively modest ~1.6&times;10&#8315;&#8310; — still significant, but far closer to the noise floor than the other two. A weak-but-significant score doesn't mean "probably not there"; it means the profile HMM found less redundant, less strongly conserved sequence evidence to key on, which is the kind of judgment call profile-based tools like Pfam are built to make explicit rather than hide behind a single pass/fail cutoff.